WalletWasabi/Blockchain/Transactions/TransactionFactory.cs

```diff
@@ -65,17 +65,13 @@ public TransactionFactory(Network network, KeyManager keyManager, ICoinsView coi
 		}
 
 		// Get allowed coins to spend.
-		var availableCoinsView = Coins.Unspent();
+		var availableCoinsView = Coins.Available().Unspent();
 		List<SmartCoin> allowedSmartCoinInputs = AllowUnconfirmed // Inputs that can be used to build the transaction.
 				? availableCoinsView.ToList()
 				: availableCoinsView.Confirmed().ToList();
-		if (allowedInputs is not null) // If allowedInputs are specified then select the coins from them.
-		{
-			if (!allowedInputs.Any())
-			{
-				throw new ArgumentException($"{nameof(allowedInputs)} is not null, but empty.");
-			}
```

Comment:
I don't get this "allowedInputs is not null, but empty" exception.

I can revert this if there is a good reason for it.

Reply:
Tests fails without this, so reverting.